doc/devel/testing.rst

```diff
@@ -144,13 +144,68 @@ tests it::
        fig, ax = plt.subplots()
        ax.plot(range(10), linestyle=(0, (3, 3)), lw=5)
 
-The first time this test is run, there will be no baseline image to compare
-against, so the test will fail.  Copy the output images (in this case
-:file:`result_images/test_lines/test_line_dashes.png`) to the correct
-subdirectory of :file:`baseline_images` tree in the source directory (in this
-case :file:`lib/matplotlib/tests/baseline_images/test_lines`).  Put this new
-file under source code revision control (with ``git add``).  When rerunning
-the tests, they should now pass.
+Baseline image generation
+-------------------------
+The first time this test is run, there will be no baseline image to compare against, so the test will fail. The baseline
+images would be generated either by installing ``matplotlib-baseline-images`` package or by running the baseline image
+generation command from the commandline. These methods are described below. Later on, developer can run ::
+
+    python3 -pytest
+
+and the tests will pass.
+
+
+How to install matplotlib-baseline-images package?
+--------------------------------------------------
+Developer can install the ``matplotlib-baseline-images`` package by running the command ::
+
+    python3 -mpip install  -ve  sub-wheels/matplotlib-baseline-images
+
+
+Baseline image generation at the fresh install of matplotlib through command line
+----------------------------------------------------------------------------------
+The baseline images for the :file:`lib/matplotlib` directory can be generated by using ::
+
+    python3 -mpytest lib/matplotlib --generate_images
+
+This will create baseline images in the respective ``baseline_images`` directory.
+Similarly, the baseline images for the :file:`lib/mpl_toolkits` directory can be generated by using ::
+
+    python3 -mpytest lib/mpl_toolkits --generate_images
+
+This will create baseline images in the respective  :file:`lib/baseline_images` directory for the
+:file:`lib/mpl_toolkits` folder.
+
+
+How to generate the modified baseline images?
+---------------------------------------------
+Tests generating the baseline images may change over time. So, the baseline images needs to be regenerated by the
```

Comment:
This probably should also be in the contributing section as this is something that developers will have to do regularly?